OptimizelySDK.Tests/OptimizelyTest.cs

```diff
@@ -1109,5 +1120,350 @@ public void TestGetVariationBucketingIdAttribute()
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns<string>(null);
+
+            Assert.AreEqual(100.54, OptimizelyMock.Object.GetFeatureVariableDouble(featureKey, variableKeyDouble, TestUserId, null));
+            Assert.AreEqual(100, OptimizelyMock.Object.GetFeatureVariableDouble(featureKey, variableKeyInt, TestUserId, null));
+            Assert.AreEqual(null, OptimizelyMock.Object.GetFeatureVariableDouble(featureKey, variableKeyNonDouble, TestUserId, null));
+            Assert.AreEqual(null, OptimizelyMock.Object.GetFeatureVariableDouble(featureKey, variableKeyNull, TestUserId, null));
+        }
+
+        [Test]
+        public void TestGetFeatureVariableIntegerReturnTypecastedValue()
+        {
+            var featureKey = "featureKey";
+            var variableKeyInt = "varInt";
+            var variableKeyDouble = "varDouble";
+            var variableNonInt = "varNonInt";
+            var variableKeyNull = "varNull";
+            var featureVariableType = FeatureVariable.VariableType.INTEGER;
+
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyInt, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("100");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyDouble, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("100.45");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableNonInt, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("non_integer_value");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyNull, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns<string>(null);
+
+            Assert.AreEqual(100, OptimizelyMock.Object.GetFeatureVariableInteger(featureKey, variableKeyInt, TestUserId, null));
+            Assert.AreEqual(null, OptimizelyMock.Object.GetFeatureVariableInteger(featureKey, variableKeyDouble, TestUserId, null));
+            Assert.AreEqual(null, OptimizelyMock.Object.GetFeatureVariableInteger(featureKey, variableNonInt, TestUserId, null));
+            Assert.AreEqual(null, OptimizelyMock.Object.GetFeatureVariableInteger(featureKey, variableKeyNull, TestUserId, null));
+        }
+
+        [Test]
+        public void TestGetFeatureVariableStringReturnTypecastedValue()
+        {
+            var featureKey = "featureKey";
+            var variableKeyString = "varString1";
+            var variableKeyIntString = "varString2";
+            var variableKeyNull = "varNull";
+            var featureVariableType = FeatureVariable.VariableType.STRING;
+
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyString, It.IsAny<string>(),
```

Comment:
Please re-arrange the order of getFeatureVariableValueForType. userAttributes should come at last.

Reply:
As per doc and python, variableType comes last